[HPB-161] [FE] 建立 CRUD API 層

richardlinio opened the pull request:
[Issue Link](https://www.notion.so/jyu1999/FE-CRUD-API-278cba948f11809587e5de79ebfd1470?source=copy_link)

## Summary

新增 `useArticle` composable 統一管理文章相關的 API 操作，封裝對文章的 CRUD 邏輯，並提供了單元測試確保其穩定性與正確性。

## Major changes

- **`app/composables/useArticle.ts`**:
  - 建立了一個新的 composable `useArticle` 來處理所有與文章相關的資料互動。
  - 提供了 `list`, `get`, `create`, `update`, `remove` 等非同步函式，分別對應文章列表、單一文章查詢、新增、更新與刪除操作。
  - 整合了 `useApiClient`，能自動處理 API request 所需的認證 token。
  - 內部管理了 `articles` 列表、`currentArticle`（當前文章）、`loading`（載入狀態）和 `error`（錯誤狀態）等響應式狀態（reactive state），簡化了在 Vue component 中的使用。

- **`test/nuxt/useArticle.test.ts`**:
  - 為新的 `useArticle` composable 撰寫了完整的 `vitest` 單元測試。
  - 透過模擬 (mock) `useApiClient` 的方式來隔離測試，確保測試的重點在於 `useArticle` 本身的商業邏輯，而非外部 API client。
  - 測試案例涵蓋了所有函式的成功情境、錯誤處理、載入狀態的變化，以及在列表更新、併發請求等各種邊界情況下的行為。

## How to test

請執行以下指令，並確認所有測試案例皆通過，且沒有任何錯誤或警告訊息。

```bash
make ci
```

richardlinio committed:
feat(useArticle): add article management composable with CRUD operations
- implement list, get, create, update, and remove functions for articles
- manage loading and error states for API calls
- update articles list and current article reactive references accordingly
- integrate with useApiClient for API requests

test(useArticle): add comprehensive tests for article composable
- mock useApiClient methods for isolated testing
- test list, get, create, update, and remove functionalities including success and error cases
- verify reactive state updates for articles list, currentArticle, loading, and error

richardlinio committed:
refactor(useArticle): remove unused params argument from list function and update tests accordingly
- remove params argument from list to simplify API call
- update apiClient.get call to exclude params
- modify tests to remove query params usage and expectations
- remove redundant test for list without query params

richardlinio committed:
style(useArticle.ts): translate comments from Chinese to English
- Translate comments for create, update, and delete article operations to English
- Improve code readability with consistent comment language

richardlinio committed:
test(useArticle): add loading state, error clearing, concurrency, and empty list tests
- add loading state tests for get, create, update, and remove operations including error cases
- add tests ensuring articles list and currentArticle are not updated on create/update/remove failures
- add tests for clearing previous errors on successful operations after failures
- add tests for handling concurrent list and get calls, verifying last resolved wins
- add tests for operations when articles list is empty, ensuring correct behavior

richardlinio committed:
refactor(useArticle): replace ref with useState for reactive state management
- change articles, currentArticle, loading, and error from ref to useState in useArticle composable
- update state initialization with useState to enable shared state across components

test(useArticle): reset useState values before each test
- clear articles, currentArticle, loading, and error state values in useArticle test setup to avoid state leakage between tests

richardlinio committed:
perf(Makefile): run lint and typecheck commands in parallel for faster execution
- Run eslint and nuxt typecheck concurrently in lint and lint-verbose targets
- Execute test and lint-verbose targets in parallel during ci target

richardlinio committed:
refactor(Makefile, CLAUDE.md): add llm-specific make targets for CI pipeline and update documentation
- Add fmt-llm, lint-llm, test-llm, and ci-llm targets in Makefile with quieter output filtering
- Rename existing format, lint, test, and ci targets to non-llm versions (fmt, lint, test, ci)
- Update CLAUDE.md to document new llm-specific make targets and replace old target names accordingly

richardlinio commented on Makefile:
```diff
@@ -1,23 +1,38 @@
+fmt-llm:
 	@npx eslint . --fix >/dev/null 2>&1 || true
 	@npx prettier --write . >/dev/null 2>&1 || true
 
 lint:
-	@npx eslint . --quiet
-	@npx nuxt typecheck --logLevel=silent
+	@(npx eslint .) & (npx nuxt typecheck) & wait
 
-lint-verbose:
-	@npx eslint .
-	@npx nuxt typecheck
+lint-llm:
+	@set -o pipefail; \
+	(npx eslint . 2>&1 | grep -v -E '^\[nuxt\]|\[info\]|^ℹ|\[nitro\]') & \
+	(npx nuxt typecheck 2>&1 | grep -v -E '^\[nuxt\]|\[info\]|^ℹ|\[nitro\]|.*\[.*ms\]|.*↳|@tailwindcss') & \
+	wait
 
 test:
-	@npx vitest run --reporter=dot --silent
+	@npx vitest run 
+
+test-llm:
+	@set -o pipefail; npx vitest run --reporter=dot 2>&1 | \
+		grep -v -E '^\[nuxt\]|\[info\]|^ℹ|\[nitro\]|.*\[.*ms\]|.*↳|\(node:.*\) Warning:|Use.*--trace-warnings|^·+$$|@tailwindcss'
 
 ci:
 	@echo "Running format..."
-	@$(MAKE) format || true
+	@$(MAKE) fmt
 	@echo "Running tests and lint in parallel..."
-	@($(MAKE) test || true) & ($(MAKE) lint-verbose || true) & wait
-	@echo "CI pipeline completed."
\ No newline at end of file
+	@$(MAKE) test & $(MAKE) lint & wait
+	@echo "CI pipeline completed."
+
+ci-llm:
+	@$(MAKE) fmt-llm
+	@$(MAKE) test-llm & $(MAKE) lint-llm & wait
```
不用 review，只是調整一些我個人降低輸出、最小化 LLM token 使用量的指令，不影響其他程式碼或 CI 流程

KR0924 commented on app/composables/useArticle.ts:
```diff
@@ -0,0 +1,118 @@
+    }
+  }
+
+  async function create(article: NewArticle) {
```
這參數能換成newArticle，我讀到下面有時候會需要回來看他是什麼屬性

richardlinio replied:
同意，已改名 newArticle
已調整於 commit: https://github.com/Goodidea-backend-camp/hpb-blog-frontend/pull/7/commits/054ba10d0d9cafa40f803c7c60e2b15d86211158

KR0924 commented on app/composables/useArticle.ts:
```diff
@@ -0,0 +1,118 @@
+    }
+  }
+
+  async function update(slug: string, article: UpdateArticle) {
```
問題同[這邊](https://github.com/Goodidea-backend-camp/hpb-blog-frontend/pull/7/files#r2415253575)

richardlinio replied:
同意，已改名 updatedArticle
已調整於 commit: https://github.com/Goodidea-backend-camp/hpb-blog-frontend/pull/7/commits/054ba10d0d9cafa40f803c7c60e2b15d86211158

richardlinio committed:
refactor(useArticle): rename function parameters for clarity

- Rename parameter `article` to `newArticle` in `create` function
- Rename parameter `article` to `updatedArticle` in `update` function
